src/formatDate.js

```diff
@@ -8,7 +8,49 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const separator = toFormat[3];
```

Comment:
The assumption that the separator is always at index 3 in the format arrays may not be valid. Consider dynamically determining the separator by checking for non-alphanumeric characters in the format strings.